assets/src/components/form/model/base/model-select.js

```diff
@@ -83,7 +84,24 @@ export class ModelSelect extends Component {
 		getQueryString: () => '',
 	};
 
-	static getDerivedStateFromProps( props ) {
+	constructor( props ) {
+		super( props );
+		this.state = ModelSelect.setStateFromProps( props );
+	}
```

Comment:
state is initialized on Component construction (important for setting defaults to expected values)